pkg/kubelet/kubelet.go

```diff
@@ -2275,10 +2275,11 @@ func (kl *Kubelet) deletePod(pod *v1.Pod) error {
 // and updates the pod to the failed phase in the status manager.
 func (kl *Kubelet) rejectPod(pod *v1.Pod, reason, message string) {
 	kl.recorder.Eventf(pod, v1.EventTypeWarning, reason, message)
-	kl.statusManager.SetPodStatus(pod, v1.PodStatus{
-		Phase:   v1.PodFailed,
-		Reason:  reason,
-		Message: "Pod was rejected: " + message})
+	status := pod.Status.DeepCopy()
```

Comment:
hm, I am a worried about this deep copy. Let's be explicit what we want to preserve here, code that only was setting QoS was OK. The test will catch issue when some other fields will need to be added here.

Reply:
the new test is failing. should I remove the condition from the expected status?

```
  [FAILED] Value for field 'Status' failed to satisfy matcher.
  Expected
      <v1.PodStatus>:
          message: 'Pod was rejected: Node didn''t have enough resource: cpu, requested: 1000000000000000,
            used: 950, capacity: 8000'
          phase: Failed
          qosClass: Burstable
          reason: OutOfcpu
          startTime: "2024-10-19T10:18:01Z"
  to equal
      <v1.PodStatus>:
          conditions:
          - lastProbeTime: null
            lastTransitionTime: "2024-10-19T10:17:59Z"
            message: '0/1 nodes are available: 1 Insufficient cpu. preemption: 0/1 nodes are
              available: 1 No preemption victims found for incoming pod.'
            reason: Unschedulable
            status: "False"
            type: PodScheduled
          message: 'Pod was rejected: Node didn''t have enough resource: cpu, requested: 1000000000000000,
            used: 950, capacity: 8000'
          phase: Failed
          qosClass: Burstable
          reason: OutOfcpu
          startTime: "2024-10-19T10:18:01Z"
  In [It] at: /Users/kiki/workspace/golang/src/k8s.io/kubernetes/test/e2e/common/node/pod_admission.go:140
```

Reply:
> code that only was setting QoS was OK. 

I updated the e2e test to match the kubelet changes.